nemo/collections/asr/data/audio_to_text_dataset.py

```diff
@@ -398,6 +400,88 @@ def get_tarred_dataset(
     return get_chain_dataset(datasets=datasets, ds_config=config, rank=global_rank)
 
 
+def get_code_switched_dataset(
+    config: dict,
+    shuffle_n: int,
+    global_rank: int,
+    world_size: int,
+    tokenizer: Optional['TokenizerSpec'] = None,
+    augmentor: Optional['AudioAugmentor'] = None,
+) -> CodeSwitchedDataset:
+
+    tarred_audio_filepaths = config.get('tarred_audio_filepaths', None)
+    manifest_filepaths = config['manifest_filepath']
+
+    # needed to support validation Datasets that arrive here as
+    # [[dataset1,dataset2]] otherwise ModelPT would interfere
+    if len(manifest_filepaths) == 1 and not isinstance(manifest_filepaths[0], str):
+        manifest_filepaths = config['manifest_filepath'][0]
+    if tarred_audio_filepaths is None:
+        tarred_audio_filepaths = [None] * len(manifest_filepaths)
+    datasets = {}
+    for dataset_idx, (tarred_audio_filepath, manifest_filepath) in enumerate(
+        zip(tarred_audio_filepaths, manifest_filepaths)
+    ):
+        lang = config['code_switch_languages'][dataset_idx]
+        conf = copy.deepcopy(config)
+        conf['manifest_filepath'] = manifest_filepath
+        with open_dict(conf):
+            conf['tarred_audio_filepaths'] = tarred_audio_filepath
+            conf['language'] = lang
+        if tarred_audio_filepath is None or len(tarred_audio_filepath) == 0:
+            if tokenizer is None:
+                dataset = get_char_dataset(config=conf, augmentor=None)
+            else:
+                dataset = get_bpe_dataset(config=conf, tokenizer=tokenizer, augmentor=None)
+        else:
+            dataset = get_tarred_dataset(
+                config=conf,
+                tokenizer=tokenizer,
+                shuffle_n=shuffle_n,
+                global_rank=global_rank,
+                world_size=world_size,
+                augmentor=None,
```

Comment:
Actually this is by design. We don't want to pass the augmentor to the individual dataset classes because we then sample from them and we want un-augmented audio returned to build the synthetic utterance with. The augmentor, if provided, is only used on the final, composite code-switched synthetic sample. This is why it's always None until you instantiate the final code_switched_dataset object and only then do you pass the augmentor to that object.